packages/eslint-plugin/src/rules/prefer-function-type.ts

```diff
@@ -26,6 +33,35 @@ export default util.createRule({
   create(context) {
     const sourceCode = context.getSourceCode();
 
+    /**
+     * Get the range of the `this` which is being used as a type annotation else returns null
+     * @param node The node being checked
+     * @returns {Array<number> | null} the range or null if no `this` type annotation are found
+     */
+    function getReturnType(
+      node: TSESTree.TSTypeAnnotation,
+    ): Array<number> | null {
+      if (!possibleReturnThisTypeHolders.has(node.type)) {
+        return null;
+      }
+
+      if (node.type === AST_NODE_TYPES.TSThisType) {
+        return node.range;
+      }
```

Comment:
> There are few overlapping issue while comparing which is why I think the CI is failing, should I turn off the rule there ?

This is failing the typecheck because it can never be true.
Your argument is of type `TSESTree.TypeAnnotation`, which is a container node.

See my other comment for a suggestion of how to structure the function.